src/assertions.ts

```diff
@@ -366,6 +369,81 @@ export async function runAssertion({
     };
   }
 
+  if (baseType === 'is-sql') {
+    let parsedSQL;
+    let databaseType: string | undefined = 'MySQL';
+    let whiteTableList: string[] | undefined;
+    let whiteColumnList: string[] | undefined;
+
+    interface sqlParserOption {
+      database: string | undefined;
+      type?: string; // 'type' is optional
+    }
+
+    if (renderedValue) {
+      if (typeof renderedValue === 'object') {
+        const value = renderedValue as {
+          database?: string;
+          whiteTableList?: string[];
+          whiteColumnList?: string[];
+        };
+
+        databaseType = value.database || 'MySQL';
+        whiteTableList = value.whiteTableList;
+        whiteColumnList = value.whiteColumnList;
+      } else {
+        throw new Error('is-sql assertion must have a object value.');
+      }
+    }
+
+    let opt: sqlParserOption = { database: databaseType };
+    let failed_reason = '';
```

Comment:
consider defining 
```suggestion
    const failureReasons = [];
```
then pushing to the array, and joining at the end. It will solve your leading whitespace problems with some of the error messages.

Reply:
got it!